syncapi/synctypes/clientevent.go

```diff
@@ -71,28 +101,57 @@ func ToClientEvents(serverEvs []gomatrixserverlib.PDU, format ClientEventFormat,
 				sk = &skString
 			}
 		}
-		evs = append(evs, ToClientEvent(se, format, sender, sk))
+
+		unsigned := se.Unsigned()
+		var prev PrevEventRef
+		if err := json.Unmarshal(se.Unsigned(), &prev); err == nil && prev.PrevSenderID != "" {
+			prevUserID, err := userIDForSender(*validRoomID, spec.SenderID(prev.PrevSenderID))
+			if err == nil && userID != nil {
+				prev.PrevSenderID = prevUserID.String()
+			} else {
+				logrus.Warnf("Failed to find userID for prev_sender in ClientEvent")
+				// NOTE: Not much can be done here, so leave the previous value in place.
```

Comment:
We are dropping `err` to the floor here without logging in the case where `err != nil` and `userID == nil`, is this intentional?